### ISSUE TITLE
Test loss doesn't go down

### ISSUE THREAD

**Issue by bryanyzhu** [yjxiong/caffe#46]
Hi,

Thanks for the great work, I am using it to fine tune on another action recognition dataset. But right now the situation is the training loss is going down very slowly, but the test loss remain the same. Specifically, starting from iteration 5000, the training loss is 2+ and the test loss is 2+. Now the training loss is around 1, but test loss is still 2+. Could anyone point out any issue or any suggestions? I am using dropout ratio of 0.5 and step size 10000, learning rate start from 0.001. I think maybe overfitting, but I got more than 50 million frames, which shouldn't overfit so quickly.  

I0218 15:34:31.817140 38066 solver.cpp:240] Iteration 17680, loss = 0.913562
I0218 15:34:31.817322 38066 solver.cpp:631] Iteration 17680, lr = 0.0001
I0218 15:37:39.009559 38066 solver.cpp:240] Iteration 17700, loss = 0.984204
I0218 15:37:39.009740 38066 solver.cpp:631] Iteration 17700, lr = 0.0001
I0218 15:40:46.215205 38066 solver.cpp:240] Iteration 17720, loss = 0.88048
I0218 15:40:46.215389 38066 solver.cpp:631] Iteration 17720, lr = 0.0001
I0218 15:43:53.435811 38066 solver.cpp:240] Iteration 17740, loss = 0.950506
I0218 15:43:53.435973 38066 solver.cpp:631] Iteration 17740, lr = 0.0001
I0218 15:47:00.659432 38066 solver.cpp:240] Iteration 17760, loss = 0.954575
I0218 15:47:00.659607 38066 solver.cpp:631] Iteration 17760, lr = 0.0001
I0218 15:50:07.868850 38066 solver.cpp:240] Iteration 17780, loss = 0.931296
I0218 15:50:07.869019 38066 solver.cpp:631] Iteration 17780, lr = 0.0001
I0218 15:53:15.080535 38066 solver.cpp:240] Iteration 17800, loss = 0.978098
I0218 15:53:15.080703 38066 solver.cpp:631] Iteration 17800, lr = 0.0001
I0218 15:56:22.298936 38066 solver.cpp:240] Iteration 17820, loss = 0.919493
I0218 15:56:22.299123 38066 solver.cpp:631] Iteration 17820, lr = 0.0001
I0218 15:59:29.498234 38066 solver.cpp:240] Iteration 17840, loss = 0.845611
I0218 15:59:29.498405 38066 solver.cpp:631] Iteration 17840, lr = 0.0001
I0218 16:02:36.685513 38066 solver.cpp:240] Iteration 17860, loss = 0.95367
I0218 16:02:36.685740 38066 solver.cpp:631] Iteration 17860, lr = 0.0001
I0218 16:05:43.915957 38066 solver.cpp:240] Iteration 17880, loss = 0.83995
I0218 16:05:43.916149 38066 solver.cpp:631] Iteration 17880, lr = 0.0001
I0218 16:08:51.107620 38066 solver.cpp:240] Iteration 17900, loss = 0.891882
I0218 16:08:51.107802 38066 solver.cpp:631] Iteration 17900, lr = 0.0001
I0218 16:11:58.317132 38066 solver.cpp:240] Iteration 17920, loss = 1.18598
I0218 16:11:58.317296 38066 solver.cpp:631] Iteration 17920, lr = 0.0001
I0218 16:15:05.533668 38066 solver.cpp:240] Iteration 17940, loss = 1.07879
I0218 16:15:05.533854 38066 solver.cpp:631] Iteration 17940, lr = 0.0001
I0218 16:18:12.735044 38066 solver.cpp:240] Iteration 17960, loss = 1.05319
I0218 16:18:12.735213 38066 solver.cpp:631] Iteration 17960, lr = 0.0001
I0218 16:21:19.916901 38066 solver.cpp:240] Iteration 17980, loss = 0.985989
I0218 16:21:19.917068 38066 solver.cpp:631] Iteration 17980, lr = 0.0001
I0218 16:24:19.163748 38066 solver.cpp:502] Snapshotting to action_vgg_16_rgb_iter_18000.caffemodel
I0218 16:24:27.496953 38066 solver.cpp:510] Snapshotting solver state to action_vgg_16_rgb_iter_18000.solverstate
I0218 16:24:32.482759 38066 solver.cpp:335] Model Synchronization Communication time 0.27824 second
I0218 16:24:32.482831 38066 solver.cpp:428] Iteration 18000, Testing net (#0)
I0218 16:25:11.642127 38066 solver.cpp:484]     Test net output #0: accuracy = 0.471176
I0218 16:25:11.642263 38066 solver.cpp:484]     Test net output #1: loss = 2.31355 (\* 1 = 2.31355 loss)
I0218 16:25:20.908124 38066 solver.cpp:240] Iteration 18000, loss = 1.11894
I0218 16:25:20.908184 38066 solver.cpp:631] Iteration 18000, lr = 0.0001
I0218 16:28:27.909950 38066 solver.cpp:240] Iteration 18020, loss = 0.86253
I0218 16:28:27.910153 38066 solver.cpp:631] Iteration 18020, lr = 0.0001
I0218 16:31:35.093183 38066 solver.cpp:240] Iteration 18040, loss = 0.868103

**PS:** I also try dropout ratio 0.9, step size 4000 as in the UCF101 example. Around iteration 10000, both training loss and test loss is around 2.0, and doesn't go down. Is there a way to pass this plateau? Thanks

**Comment by yjxiong** [comment]
I would like to help, but I cannot tell too much from only changes of the loss values. It depends on many factors, like number of classes, size of the dataset, and also your preprocessing choices. 
The only point I can think of right now is that the step size may be small for your dataset if it is larger than UCF101.

**Comment by bryanyzhu** [comment]
Hi @yjxiong  thanks for your reply, I am doing experiment on Activity Net dataset, release 1.2. So the dataset have 100 classes, 4000+ training videos, 2000+ validation and 2000+ test videos. I would say approximately the average number of frames of each videos is around 3000. So the situation is I have 7000 train+val videos, and 7000*3000 (more than 20 million) RGB frames.

I just use the preprocessing methods in your UCF101 project, didn't change anything. The only change I made is set training batch size to 8 because the memory limit of my GPU card. So the training batch size is 8, test batch size is 5, 4 GPUS, iter size is set to 8. 

I also think maybe step size of 4000 is small, so I changed to 10000, maybe larger? 

I still have a simple question, so if I have 20 million frames, the batch size is 8x4x8 = 256, then I need at least 80K iterations to finish one epoch, do I understand correct? Thanks.

**Comment by yjxiong** [comment]
I think for video it is better to measure the volume of a data set by the number of clips rather than frames. In the sense that frames within a same clip have less variances than between the clips, please try consider the clip number in tuning the experiment configs.

**Comment by bryanyzhu** [comment]
Thanks a lot for your suggestions @yjxiong So If i want to reduce overfitting, I can simply try reduce learning rate, more data augmentation, high dropout ratio, etc. anything else? I appreciate your help.

And for the test_iter =170 in the solver file, for a dataset larger than UCF101, I also need to increase this value, right? Thanks.

**Comment by yjxiong** [comment]
Yes you need to make sure your validation process covers the whole Val set. 

For overfitting, these you mentioned are instant approaches you may try. But more attention has to be paid to the problem you are dealing with.

**Comment by bryanyzhu** [comment]
@yjxiong sure thanks.
